src/js/core/UpdateStream.js

```diff
@@ -0,0 +1,100 @@
+import { Observable } from "rxjs/Observable";
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import { request } from "@dcos/http-service";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+import compareVersions from "compare-versions";
+
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION = "dcosUIVersion";
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+
+function getFromLocalStorage(key) {
+  const value = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    DCOS_UI_VERSION + "." + key
+  );
+
+  return value;
+}
+
+function setInLocalStorage(key, value) {
+  const savedStates = UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates[DCOS_UI_VERSION] = {
+    dismissedVersion: getFromLocalStorage("dismissedVersion") || null,
+    lastTimeCheck: getFromLocalStorage("lastTimeCheck") || null
+  };
+  savedStates[DCOS_UI_VERSION][key] = value;
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+}
+
+function getContentType({ action, actionType, entity, version }) {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+function showNotification(newVersion) {
+  // TODO: show notification
+  // When dismissing, save dismissed version in local storage
+  const currentVersion = window.DCOS_UI_VERSION;
+  console.log(
+    "A new version of DCOS UI is available. Upgrade " +
+      currentVersion +
+      " to " +
+      newVersion +
+      ". Dismiss."
+  );
+}
+
+setInLocalStorage("dismissedVersion", "2.24.2"); // stub
+const dismissedVersion = Observable.of(
```

Comment:
We can use `startsWith` for default values http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-startWith

Reply:
I think by declaring it like that we effectively pin the version that was in the localStore on browser load and never re-check it. 

Why not making `getFromLocalStorage` observable with `Observable.create`?